### PR TITLE
Describe features at a high level rather than querying the user model directly

### PR DESCRIPTION
**Pull request by johngallagher**
## What

The platform team have recently added a `Features::FeatureRepo` that contains all feature definitions.

Let's use this for any features we need to define in future.

## Why

So that we can have consistency in how we define features and be able to see them all in one place.

### CODE REVIEW

**Review by johngallagher — COMMENTED** [review]
Need to add more explanations for how `enabled?` works and also how `value` works.

File: _patterns/use-feature-repo-for-feature-access-checks.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    # ...
+  end
+  # ...
+end
+```
+
+Then in the `Features` module:
+
+```ruby
+# lib/features/feature_repo.rb
```
Comment (karlentwistle) [inline _patterns/use-feature-repo-for-feature-access-checks.md:67]: I think this part needs to be updated with `lib/features/features.rb` instead

**Comment by johngallagher** [comment]
@baerbradford just updated this as discussed to be more generic. Let me know if there's anything else you think is too risky to expose.

File: _patterns/use-feature-repo-for-feature-access-checks.md (hunk 1 above)
Comment (johngallagher) [reply _patterns/use-feature-repo-for-feature-access-checks.md]: I've eliminated all the paths as per @baerbradford 's concerns

**Review by baerbradford — APPROVED** [review]

**Review by RajaaBoulassouak — APPROVED** [review]

**Review by RenatoCoelho-BiggerPockets — APPROVED** [review]

**Review by kxhitiz — APPROVED** [review]

**Review by cernivec — APPROVED** [review]

**Pull request merged**